employee-rostering-distribution/pom.xml

```diff
@@ -103,6 +122,75 @@
           </execution>
         </executions>
       </plugin>
+      <plugin> <!-- Start the application as a separate process -->
+        <groupId>com.bazaarvoice.maven.plugins</groupId>
+        <artifactId>process-exec-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>start-backend-service</id>
+            <phase>pre-integration-test</phase>
+            <goals>
+              <goal>start</goal>
+            </goals>
+            <configuration>
+              <name>Run application</name>
+              <healthcheckUrl>http://localhost:${application.port}</healthcheckUrl>
+              <workingDir>${project.build.directory}</workingDir>
+              <processLogFile>${project.build.directory}/${project.artifactId}.log</processLogFile>
+              <arguments>
+                <argument>java</argument>
+                <argument>-Dspring.profiles.active=dev</argument>
+                <argument>-Dserver.port=${application.port}</argument>
+                <argument>-jar</argument>
+                <argument>${project.artifactId}-${project.version}.jar</argument>
+              </arguments>
+            </configuration>
+          </execution>
+          <!-- Kill all running processes -->
+          <execution>
+            <id>stop-running-processes</id>
+            <phase>post-integration-test</phase>
+            <goals>
+              <goal>stop-all</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <!-- Run Cypress tests in a docker container.
+               See https://www.cypress.io/blog/2019/05/02/run-cypress-with-a-single-docker-command -->
+      <plugin>
+        <artifactId>exec-maven-plugin</artifactId>
+        <groupId>org.codehaus.mojo</groupId>
+        <executions>
+          <execution>
+            <id>run-cypress-tests</id>
+            <phase>integration-test</phase>
+            <goals>
+              <goal>exec</goal>
+            </goals>
+            <configuration>
+              <skip>${skipITs}</skip>
+              <executable>docker</executable>
```

Comment:
I am assuming the test will fail if we don't have docker installed on our system? (I prefer `podman`, which I have a `docker` alias for)

Reply:
Correct - and this is exactly the behaviour I count on in the CI. On some machines there might be docker, on others there is podman and docker alias for it.

Reply:
Might want to add "if you don't have `docker` or `podman` with a `docker` alias installed on your system, pass `-DskipITs` to maven" so a developer without `docker`/`podman` (they exist) doesn't get confused about why IT tests are failing.

Reply:
(to `README.md`); does `-DskipTests` imply `DskipITs`?

Reply:
No, `skipITs` is reserved for integration tests (e.g. by maven failsafe plugin). I noticed that in the README.md we skipTests, so I can add skipITs there too.

Reply:
Maybe use [fabric8io/docker-maven-plugin](https://dmp.fabric8.io/) or https://github.com/spotify/docker-maven-plugin instead of exec?

Reply:
Unfortunately the plugin required docker service to be running. Instead, I made the container runtime we use configurable, so that it's possible to use podman as well.

@yurloc thanks a lot for testing this with a podman and a docker alias.